### PR TITLE
Add C++ DeploymentManager sample and add ForceDeploy option

### PR DESCRIPTION
**Pull request by angelazhangmsft**
## Description

- Create C++ version of DeploymentManager sample
- Update DeploymentManager C# sample with ForceDeployment option. 
- Update the sample to use the WinUI3 template

## Target Release

1.1 GA

## Checklist

- [x] Samples build and run using the Visual Studio versions listed in the [Windows development docs](https://docs.microsoft.com/windows/apps/windows-app-sdk/set-up-your-development-environment?tabs=stable#2-install-visual-studio).
- [x] Samples build and run on all supported platforms (x64, x86, ARM64) and configurations (Debug, Release).
- [x] Samples set the minimum supported OS version to Windows 10 version 1809.
- [ ] Samples build clean with no warnings or errors.
- [x] **[For new samples]**: Samples have completed the [sample guidelines checklist](https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/docs/samples-guidelines.md#checklist) and follow [standardization/naming guidelines](https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/docs/samples-guidelines.md#standardization-and-naming).
- [x] I have commented on my PR `/azp run` to have the CI build run on my branch. This must be done right before merging to ensure the build is up to date and accurate.

### CODE REVIEW

File: Samples/DeploymentManager/cs-winui/DeploymentManagerSample.csproj [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    <Platforms>x86;x64;arm64</Platforms>
+    <RuntimeIdentifiers>win10-x86;win10-x64;win10-arm64</RuntimeIdentifiers>
+    <UseWinUI>true</UseWinUI>
+    <EnablePreviewMsixTooling>true</EnablePreviewMsixTooling>
+    <PublishProfile>Properties\PublishProfiles\win10-$(Platform).pubxml</PublishProfile>
+	<WindowsSdkPackageVersion>10.0.19041.24</WindowsSdkPackageVersion>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.1.0-20220406.0-CI-experimental" />
```
Comment (angelazhangmsft) [inline Samples/DeploymentManager/cs-winui/DeploymentManagerSample.csproj:17]: update before merging

File: Samples/DeploymentManager/cpp-winui/Scenario2_Initialize.xaml.cpp [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        UpdateInitializeMessages();
+
+        if (DeploymentManager::GetStatus().Status() != DeploymentStatus::Ok)
+        {
+            // Initialize does a status check, and if the status is not Ok it will attempt to get
+            // the WindowsAppRuntime into a good state by deploying packages. Unlike a simple
+            // status check, Initialize can sometimes take several seconds to deploy the packages.
+            // These should be run on a separate thread so as not to hang your app while the
+            // packages deploy. 
+            DeploymentResult deploymentResult = DeploymentManager::Initialize(); // TODO how to run on separate thread?
```
Comment (angelazhangmsft) [inline Samples/DeploymentManager/cpp-winui/Scenario2_Initialize.xaml.cpp:65]: How do I run a separate thread for C++/WinRT?

File: Samples/DeploymentManager/cpp-winui/Scenario2_Initialize.xaml.cpp [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            // with the Main and Singleton packages if they are currently in use.
+            DeploymentInitializeOptions deploymentInitializeOptions;
+            deploymentInitializeOptions.ForceDeployment(true);
+
+            // Initialize does a status check, and if the status is not Ok it will attempt to get
+            // the WindowsAppRuntime into a good state by deploying packages. Unlike a simple
+            // status check, Initialize can sometimes take several seconds to deploy the packages.
+            // These should be run on a separate thread so as not to hang your app while the
+            // packages deploy. 
+            DeploymentResult deploymentResult = DeploymentManager::Initialize(deploymentInitializeOptions); // TODO how to run on separate thread?
```
Comment (angelazhangmsft) [inline Samples/DeploymentManager/cpp-winui/Scenario2_Initialize.xaml.cpp:91]: force deployment initialize currently throws an exception (C# and C++)

**Review by sachintaMSFT — APPROVED** [review]

**Pull request merged**